### ISSUE TITLE
`opam install -w .` ignores uncommited changes

### ISSUE THREAD

**Issue by sim642** [ocaml/opam#4010]
I cloned an OCaml project with git for local development and made some changes without committing them, because I wanted to test them out in a different project.
Therefore I followed [opam 2.0 tips](https://opam.ocaml.org/blog/opam-20-tips/#Automatic-pinning) and ran `opam install -w .` in the cloned and modified project to pin the dependency to the uncommitted changes.

Even after building the dependent project completely from scratch, it didn't include the uncommitted changes in the dependency. When I committed those and ran `opam install -w .` again, the dependent project finally got the changes in the dependency.

Either I'm doing something wrong or `-w` is doing something wrong and only installing the committed changes but not the uncommitted ones, even though it's entire point is to do the latter.

---

`opam list` contains the dependency pinned to a `git+file:///...` path as expected.

`opam config report` outputs:
```# opam config report
# opam-version      2.0.4 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=ubuntu os-version=18.04
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              3
# repositories      1 (http) (default repo at 605ad4ae)
# pinned            1 (git)
# current-switch    4.06.0
```
where the single pinned package is the expected one.

---

Alternatively, manually pinning the cloned repository with `-k path` also correctly makes uncommitted changes show up in the dependent but that's besides the point.

**Comment by rjbou** [comment]
Seems more that `working-dir` is doing something wrong than you :) There was [several](https://github.com/ocaml/opam/pulls?utf8=%E2%9C%93&q=is%3Apr+working-dir) updates & fixes  on working-dir —especially on git synchronisation— that will be available in opam 2.1.0.

To check if your problem has already been resolved or no, can you reproduce & share the output of opam with verbose & debug information enabled `-vv --debug`

**Comment by rjbou** [comment]
Closing this issue, feel fre to reopen if needed.
